File: test_graphql_queries.py
#!/usr/bin/env python3

import requests
import json
import subprocess
import time

def test_graphql_queries():
    """Test GraphQL queries against the SQLite database"""
    
    print("🚀 Starting GraphQL API for testing...")
    
    # Start the application
    app_process = subprocess.Popen(
        ["dotnet", "run", "--no-build"],
        cwd="/tmp/5308f3564d2a4bd6ad9d22cebf285125",
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE,
        text=True
    )
    
    # Wait for application to start
    print("⏳ Waiting for application to start...")
    time.sleep(8)
    
    try:
        # Test queries
        queries = [
            {
                "name": "Schema Introspection",
                "query": """
                query {
                    __schema {
                        queryType {
                            name
                        }
                    }
                }
                """
            },
            {
                "name": "Get Users",
                "query": """
                query {
                    users {
                        id
                        firstName
                        lastName
                        email
                        isActive
                    }
                }
                """
            },
            {
                "name": "Get Products with Categories",
                "query": """
                query {
                    products {
                        id
                        name
                        price
                        category {
                            name
                        }
                    }
                }
                """
            }
        ]
        
        print("\n🔍 Testing GraphQL Queries...")
        print("=" * 50)
        
        for query_test in queries:
            print(f"\n📝 {query_test['name']}:")
            
            try:
                response = requests.post(
                    "http://localhost:5000/graphql",
                    json={"query": query_test["query"]},
                    headers={"Content-Type": "application/json"},
                    timeout=10
                )
                
                if response.status_code == 200:
                    data = response.json()
                    if "errors" in data:
                        print(f"   ❌ GraphQL Error: {data['errors']}")
                    else:
                        print(f"   ✅ Success!")
                        # Print a summary of the data
                        if "data" in data:
                            for key, value in data["data"].items():
                                if isinstance(value, list):
                                    print(f"      {key}: {len(value)} items")
                                else:
                                    print(f"      {key}: {value}")
                else:
                    print(f"   ❌ HTTP Error: {response.status_code}")
                    
            except requests.exceptions.RequestException as e:
                print(f"   ❌ Request failed: {e}")
        
        print("\n" + "=" * 50)
        print("✅ GraphQL testing completed!")
        
    finally:
        # Stop the application
        print("\n🛑 Stopping application...")
        app_process.terminate()
        app_process.wait(timeout=5)

if __name__ == "__main__":
    test_graphql_queries()

File: test_graphql_queries.js
const fetch = require('node-fetch');

const GRAPHQL_ENDPOINT = 'http://localhost:5001/graphql';

// Test queries
const queries = [
  {
    name: 'Featured Products (HomePage)',
    query: `
      query GetFeaturedProducts($first: Int = 8) {
        products(where: { isFeatured: { eq: true } }, first: $first) {
          nodes {
            id
            name
            description
            price
            compareAtPrice
            category {
              id
              name
            }
            images {
              id
              imageUrl
              altText
              isPrimary
            }
          }
        }
      }
    `,
    variables: { first: 8 }
  },
  {
    name: 'Products with Price Sorting (Catalog)',
    query: `
      query GetProducts(
        $where: ProductFilterInput
        $order: [ProductSortInput!]
        $first: Int
        $after: String
      ) {
        products(where: $where, order: $order, first: $first, after: $after) {
          nodes {
            id
            name
            description
            price
            compareAtPrice
            category {
              id
              name
            }
            images {
              id
              imageUrl
              altText
              isPrimary
            }
          }
          pageInfo {
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
          }
          totalCount
        }
      }
    `,
    variables: {
      where: { isActive: { eq: true } },
      order: [{ price: "ASC" }],
      first: 24
    }
  }
];

async function testQuery(query, variables, name) {
  try {
    const response = await fetch(GRAPHQL_ENDPOINT, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ query, variables })
    });

    const result = await response.json();
    
    if (result.errors) {
      console.log(`❌ ${name} failed:`);
      console.log(JSON.stringify(result.errors, null, 2));
      return false;
    } else {
      console.log(`✅ ${name} succeeded`);
      console.log(`   Products returned: ${result.data.products?.nodes?.length || 0}`);
      return true;
    }
  } catch (error) {
    console.log(`❌ ${name} failed with network error:`, error.message);
    return false;
  }
}

async function runTests() {
  console.log('Testing GraphQL queries...\n');
  
  let allPassed = true;
  
  for (const { name, query, variables } of queries) {
    const passed = await testQuery(query, variables, name);
    allPassed = allPassed && passed;
    console.log('');
  }
  
  if (allPassed) {
    console.log('🎉 All GraphQL tests passed!');
  } else {
    console.log('❌ Some GraphQL tests failed. Please check the backend logs.');
  }
}

// Note: This test requires the backend to be running
console.log('Note: Make sure the backend is running on http://localhost:5001');
console.log('Run: cd backend && dotnet run\n');

runTests();